thorlcr/activities/thdiskbase.cpp

```diff
@@ -204,11 +205,28 @@ void CWriteMasterBase::publish()
 {
     if (published) return;
     published = true;
+
+    const char *logicalFileName = dlfn.get();
+    StatsScopeId rootScope(SSTfile, 0U);
+    Owned<IStatisticGatherer> fileStats;
+    rootScope.setFileId(logicalFileName);
+    fileStats.setown(createStatisticsGatherer(SCTall, "", rootScope));
```

Comment:
not sure you need nesting as this write master knows the filename already, so think serialize stat. from each slave can be non-nested.